### ISSUE TITLE
DOI: Detect DOI in URL?

### ISSUE THREAD

**Issue by dstillman** [zotero/translators#2964]
Testing https://onlinelibrary.wiley.com/doi/full/10.7448/IAS.15.5.18440 from https://github.com/zotero/translators/issues/2963 in translation-server, the Wiley translation failed, but then translation-server fell back to using the DOI from the URL. It occurs to me that the DOI translator could just do that itself.

If only the URL matches, it can just translate that DOI rather than returning `multiple`. If there are other DOIs on the page, it can put the one from the URL at the top.

Any reason not to do this?

**Comment by adam3smith** [comment]
I thought I recalled having talked about this before. There's no reason not to do this. The DOIs aren't always cleanly delimited in URLs, though: we can probably assume with little downsides URL operators like  `?` `&` and `#` are not part of the DOI but if publishers do anything weirder than that, lookup will fail -- we just need to make sure that happens gracefully.

**Comment by dstillman** [comment]
Here's what we do in translation-server:

https://github.com/zotero/translation-server/blob/03290315ad76a06bcb5a993047172a0f90815175/src/webSession.js#L464-L471

Though, as you note, we should extend that to include `#` as well. I'll fix that.

**Comment by adam3smith** [comment]
Yeah, that makes sense. If easily doable, I'd check the translation server logs to make sure you don't get a bunch of DOI lookups failing that look suspicious (my vague recollection is that I've seen publisher add content like /full at the end of the DOI in the URL, but I don't think I have seen any examples recently.)

**Comment by adam3smith** [comment]
And to make this even worse, 10.22541/au.164848876.64808648/v1 is a valid DOI, the other two aren't...
